src/jaxsim/rbda/contacts/rigid.py

```diff
@@ -0,0 +1,509 @@
+        ):
+            BW_v_WB = data.base_velocity()
+            W_o_dot_B = BW_v_WB[0:3]
+            W_omega_WB = BW_v_WB[3:6]
+            W_v̇_WB, s̈ = js.ode.system_acceleration(
+                model=model,
+                data=data,
+                joint_forces=references.joint_force_references(model=model),
+                link_forces=references.link_forces(model=model, data=data),
+            )
+
+        # Convert the inertial-fixed base acceleration to a body-fixed base acceleration.
+        W_H_B = data.base_transform()
+        W_H_BW = W_H_B.at[0:3, 0:3].set(jnp.eye(3))
+        BW_X_W = math.Adjoint.from_transform(W_H_BW, inverse=True)
+        term1 = BW_X_W @ W_v̇_WB
+        term2 = jnp.zeros(6).at[0:3].set(jnp.cross(W_o_dot_B, W_omega_WB))
+        BW_v̇_WB = term1 - term2
+
+        BW_ν̇ = jnp.hstack([BW_v̇_WB, s̈])
+
+        return BW_ν̇
+
+    @staticmethod
+    def _linear_acceleration_of_collidable_points(
+        model: js.model.JaxSimModel,
+        data: js.data.JaxSimModelData,
+        BW_ν̇: jax.Array,
+    ) -> jax.Array:
+        with data.switch_velocity_representation(VelRepr.Mixed):
+            CW_J_WC_BW = js.contact.jacobian(
+                model=model,
+                data=data,
+                output_vel_repr=VelRepr.Mixed,
+            )
+            CW_J̇_WC_BW = js.contact.jacobian_derivative(
+                model=model,
+                data=data,
+                output_vel_repr=VelRepr.Mixed,
+            )
+
+            BW_ν = data.generalized_velocity()
+
+        CW_a_WC = jnp.vstack(CW_J̇_WC_BW) @ BW_ν̇ + jnp.vstack(CW_J_WC_BW) @ BW_ν
+        CW_a_WC = CW_a_WC.reshape(-1, 6)
+
+        return CW_a_WC[:, 0:3].squeeze()
+
+    @staticmethod
+    def _compute_baumgarte_stabilization_term(
+        inactive_collidable_points: jax.Array,
+        delta: jax.Array,
+        delta_dot: jax.Array,
+        K: jtp.Float,
+        D: jtp.Float,
+    ) -> jtp.Array:
+        def baumgarte_stabilization(
+            inactive: jtp.Bool,
+            delta: jax.Array,
+            delta_dot: jax.Array,
+            k_baumgarte: jtp.Float,
+            d_baumgarte: jtp.Float,
+        ) -> jtp.Array:
+            baumgarte_term = jax.lax.cond(
+                inactive,
+                lambda δ, δ̇, K, D: jnp.zeros(shape=(3,)),
+                lambda δ, δ̇, K, D: jnp.zeros(shape=(3,)).at[2].set(K * δ + D * δ̇),
+                *(
+                    delta,
+                    delta_dot,
+                    k_baumgarte,
+                    d_baumgarte,
+                ),
+            )
+            return baumgarte_term
+
+        baumgarte_term = jax.vmap(
+            baumgarte_stabilization, in_axes=(0, 0, 0, None, None)
+        )(
+            inactive_collidable_points,
+            delta,
+            delta_dot,
+            K,
+            D,
+        )
+
+        return baumgarte_term
+
+    @staticmethod
+    def _compute_link_forces_inertial_fixed(
```

Comment:
Can you use instead the following?

https://github.com/ami-iit/jaxsim/blob/822877eeef48bb195e504b092ccb60a4f86e13b4/src/jaxsim/api/common.py#L153

Reply:
Done in e1e0c6c9eea88e9d13b395a0a1e79bc73f620c0e